builtin/logical/database/path_config_connection.go

```diff
@@ -439,6 +458,9 @@ func (b *databaseBackend) connectionWriteHandler() framework.OperationFunc {
 }
 
 func storeConfig(ctx context.Context, storage logical.Storage, name string, config *DatabaseConfig) error {
+	if versions.IsBuiltinVersion(config.PluginVersion) {
```

Comment:
nit: might be worth adding a comment as to why we are modifying the config object's `PluginVersion`, behind the caller's back.

I see that we are calling `storeConfig()` in two spots, wonder if it might be better to do the built-in version munging from the callers instead?

- https://github.com/hashicorp/vault/blob/78373fa7d54dce1077e29528aa9238390a9f8413/builtin/logical/database/path_rotate_credentials.go#L140
- https://github.com/hashicorp/vault/blob/affd18dbe438cbb8f4b76ca35d53cf632eed5e12/builtin/logical/database/path_config_connection.go#L460

Reply:
Added a comment. My thinking on where to mutate was that this is more of a storage-layer concern, and it also stops me needing to duplicate the code + comment.

Reply:
It's still a bit odd that calling `storeConfig()` introduces this side-effect. Introducing a new function that explicitly updates the `PluginVersion` would be my preferred approach here.

Reply:
Ah I see that his has already been addressed. Thanks!

Reply:
Yeah I changed my mind and went with your suggestion while writing the tests, thanks!